pycloudlib/oci/utils.py

```diff
@@ -237,3 +337,38 @@ def parse_oci_config_from_env_vars() -> Optional[Dict[str, str]]:
             log.info("Replacing existing key_file path in OCI config")
         oci_config["key_file"] = key_file_path
     return oci_config
+
+
+def generate_create_vnic_details(
+    subnet_id: str,
+    networking_config: Optional[NetworkingConfig] = None,
+) -> oci.core.models.CreateVnicDetails:
+    """
+    Create a VNIC details object based on the primary network configuration.
+
+    Args:
+        subnet_id: The subnet id to use for the VNIC.
+        networking_config: The network configuration to use for the VNIC.
+
+    Returns:
+        vnic_details: The VNIC details object.
+    """
+    # default to IPv4 with public IP
+    # this will be used if networking_config is not provided or if set to AUTO
+    vnic_details = oci.core.models.CreateVnicDetails(  # noqa: E501
```

Comment:
it's helpful with `noqa` lines to add a brief justification for why the check is being skipped.